src/sas/qtgui/Perspectives/Fitting/ConstraintWidget.py

```diff
@@ -434,37 +434,107 @@ def onTabCellEdit(self, row, column):
-        # No check on function here - trust the user (R)
-        if function != constraint.func:
-            # This becomes rather difficult to validate now.
-            # Turn off validation for Edit Constraint
-            constraint.func = function
-            constraint.validate = False
+        model = constraint.value_ex[:constraint.value_ex.index(".")]
+        param = constraint.param
+        function = constraint.func
+        tab = self.available_tabs[model]
+        # Extract information from the text in the table
+        constraint_text = item.data(0)
+        # Basic sanity check of the string
+        # First check if we have an acceptable sign
+        if "=" not in constraint_text:
+            msg = ("Incorrect operator in constraint definition. Please use = "
+                   "sign to define constraints.")
+            QtWidgets.QMessageBox.critical(
+                self,
+                "Inconsistent constraint",
+                msg,
+                QtWidgets.QMessageBox.Ok)
+            self.initializeFitList()
+            return
+        # Then check if the parameter is correctly defined with colons
+        # separating model and parameter name
+        parameter_full_name = constraint_text[:constraint_text.index(
+            "=")].strip()
+        if ":" not in parameter_full_name:
+            msg = ("Incorrect constrained parameter definition. Please use "
+                   "colons to separate model and parameter on the rhs of the "
+                   "definition, e.g. M1:scale")
+            QtWidgets.QMessageBox.critical(
+                self,
+                "Inconsistent constraint",
+                msg,
+                QtWidgets.QMessageBox.Ok)
+            self.initializeFitList()
+            return
+        # We can parse the string
+        new_param = item.data(0)[
+                item.data(0).index(':') + 1:item.data(0).index('=')].strip()
+        new_model = item.data(0)[:item.data(0).index(':')].strip()
+        # Check that the symbol is known so we dont get an unknown tab
+        # All the conditional statements could be grouped in one or
+        # alternatively we could check with expression.py, but we would still
+        # need to do some checks to parse the string
+        symbol_dict = self.parent.parent.perspective(
+            ).getSymbolDictForConstraints()
+        if (new_model + "." + new_param) not in symbol_dict:
+            msg = ("Incorrect constrained parameter definition. Please use "
+                   "a single known parameter in the rhs of the constraint "
+                   "definition, e.g. M1:scale = M1.radius + 2")
+            QtWidgets.QMessageBox.critical(
+                self,
+                "Inconsistent constraint",
+                msg,
+                QtWidgets.QMessageBox.Ok)
+            self.initializeFitList()
+            return
+        new_function = item.data(0)[item.data(0).index('=') + 1:].strip()
```

Comment:
`.split()[1].strip()` in this case.

I haven't looked at the entire context, but you would probably be better off doing the coarse parsing earlier in the function and using the parts later in the function:
```
lhs, rhs = re.split(" *= *", item.data(0).strip(), 1)
```